Source/WTF/wtf/ProcessID.h

```diff
@@ -36,7 +36,9 @@
 namespace WTF {
 
 #if OS(WINDOWS)
-using ProcessID = int;
+using ProcessID = int; 
+#elif PLATFORM(HAIKU)
```

Comment:
@pulkomandy this is also not allowed? should i use pid_t and cast it to int32_t?

Reply:
@pulkomandy alright it is also un conventional so let me fix by casting itself 😄

Reply:
@pulkomandy am i allowed to add casting in generic file if so where can i add?

Reply:
In this case, we are forced to change generic files anyways. (ProcessID.h is also a generic file :))
So, let's add the cast, or maybe add specific calls to the decoders/encoders to encode/decode a pid_t. I don't know which solution is preferred, I will ask webkit-dev.

Reply:
I have pushed a work in progress implementation which does what I think the WebKit developers want (but I'm not sure). I did not get the "decode" part working yet. I think you can work from that (unless they changed their mind while I was sleeping).